src/kvstore/NebulaStore.cpp

```diff
@@ -76,31 +75,81 @@ bool NebulaStore::init() {
                         if (!options_.partMan_->partExist(storeSvcAddr_, spaceId, partId)) {
                             LOG(INFO) << "Part " << partId
                                       << " does not exist any more, remove it!";
                             enginePtr->removePart(partId);
                             continue;
                         } else {
-                            LOG(INFO) << "Load part " << spaceId << ", " << partId << " from disk";
-                            spaceIt->second->parts_.emplace(partId,
-                                                            newPart(spaceId,
-                                                                    partId,
-                                                                    enginePtr.get(),
-                                                                    false));
+                             partIds.emplace_back(partId);
                         }
                     }
+                    if (partIds.empty()) {
+                        continue;
+                    }
+
+                    std::atomic<size_t> counter(partIds.size());
+                    folly::Baton<true, std::atomic> baton;
+                    LOG(INFO) << "Need to open " << partIds.size() << " parts of space " << spaceId;
+                    for (auto& partId : partIds) {
+                        bgWorkers_->addTask([
+                                spaceId, partId, enginePtr, &counter, &baton, this] () mutable {
+                            auto part = std::make_shared<Part>(spaceId,
+                                                               partId,
+                                                               raftAddr_,
+                                                               folly::stringPrintf("%s/wal/%d",
+                                                                       enginePtr->getDataRoot(),
+                                                                       partId),
+                                                               enginePtr,
+                                                               ioPool_,
+                                                               bgWorkers_,
+                                                               workers_,
+                                                               snapshot_);
+                            auto partMeta = options_.partMan_->partMeta(spaceId, partId);
+                            std::vector<HostAddr> peers;
+                            for (auto& h : partMeta.peers_) {
+                                if (h != storeSvcAddr_) {
+                                    peers.emplace_back(getRaftAddr(h));
+                                    VLOG(1) << "Add peer " << peers.back();
+                                }
+                            }
+                            raftService_->addPartition(part);
+                            part->start(std::move(peers), false);
```

Comment:
I have a question，What's the difference between this code block and method newPart?

Reply:
No difference, only add a counter.

Reply:
In addition to this. LGTM! Well done！